src/utils/fxa.js

```diff
@@ -73,6 +73,41 @@ async function revokeOAuthTokens(subscriber) {
   await destroyOAuthToken({ token: subscriber.fxa_refresh_token, token_type_hint: "refresh_token" })
 }
 
+/**
+ * @param {string} refreshToken
+ * @returns {Promise<{access_token: string, refresh_token: string, expires_in: number}>}
+ */
+// Not covered by tests; mostly side-effects. See test-coverage.md#mock-heavy
+/* c8 ignore start */
+async function refreshOauthTokens(refreshToken) {
+  const subscriptionIdUrl = `${AppConstants.OAUTH_ACCOUNT_URI}/oauth/token`
+  try {
+    const postResp = await fetch(subscriptionIdUrl, {
+      headers: { "Content-Type": "application/json" },
+                body: JSON.stringify({
+                  client_id: AppConstants.OAUTH_CLIENT_ID,
+                  client_secret: AppConstants.OAUTH_CLIENT_SECRET,
+                  grant_type: "refresh_token",
+                  refresh_token: refreshToken,
+                  ttl: 604800, // request 7 days ttl
+                }),
+                method: "POST",
+              
+    })
+
+    const responseTokens = await postResp.json();
+    if (!postResp.ok) throw responseTokens;
```

Comment:
What does this line do? What would `postResp.json()` return that would be `throw`able on a non-`ok` response?

Reply:
I recently saw this as a code example on the next-auth doc. So basically it's a short hand for us to throw the correct error (.json() actually contains the error when the response isn't `ok`). In other places of our code, we'd do this whole dance of checking for `.ok` first and then do like a `.status` or `.text`. This is probably a way cleaner and better approach